src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -817,10 +827,14 @@ private void codeFieldInitialization(CodeBuilder cob) {
          * loader is anticipated at local variable index 0.
          */
         private void codeClassForName(CodeBuilder cob, Class<?> cl) {
-            cob.ldc(cl.getName())
-               .iconst_0() // false
-               .aload(0)// classLoader
-               .invokestatic(classForName);
+            if (cl == Object.class) {
+                cob.ldc(objectCE);
+            } else {
+                cob.ldc(cl.getName())
```

Comment:
This `Class.forName` is only necessary in a very small number of cases, namely when the overridden interface method has an unaccessible parameter type, usually a package-private type not accessible to the implementing class. Maybe we can always directly ldc if the class is `public`.

Reply:
Yeah, I'm just not sure what rules are quite right here. Wouldn't want to inadvertently regress this again, and not sure we have tests for all eventualities, so I started this off with the most conservative yet most beneficial improvement.